### PR TITLE
Add user agent to HTTP requests

### PR DESCRIPTION
**Pull request by ttaylorr**

### CODE REVIEW

**Comment by ttaylorr** [comment]
https://github.com/WatchBeam/beam-client-java/issues/13

File: src/main/java/pro/beam/api/http/BeamHttpClient.java [hunk 1]
```diff
@@ -186,6 +188,12 @@ private ListeningExecutorService executor() {
         return this.beam.executor;
     }
 
+    protected String getUserAgent() {
+        return new StringBuilder("BeamClient/")
+                .append(Optional.of(BeamAPI.class.getPackage().getImplementationVersion()).or("unknown"))
+                .append(" (Java, ").append(System.getProperty("java.version")).append(")").toString();
```
Comment (connor4312) [inline src/main/java/pro/beam/api/http/BeamHttpClient.java]: no comma before the version, please :smile:

**Comment by connor4312** [comment]
Can you also include the user OS? Separated by a semicolon in BeamClient comment. That's something I forgot to add to the Node version.

**Comment by ttaylorr** [comment]
> in BeamClient comment

Where?

**Comment by connor4312** [comment]
The parenthesis, with `Java`, delimited by commas

**Comment by MattJustMatt** [comment]
This PR kinda died. CC: @ttaylorr

**Comment by ttaylorr** [comment]
I revived it!

**Pull request merged**